### PR TITLE
update middleware docs

### PR DESCRIPTION
**Pull request by gurgeous**
Per the earlier discussion:

1. Copied some of the language from Usage
2. Show off more middleware and features
3. Linked to awesome-faraday (PR pending over there too)
4. Digression on `request`, `response` and `adapter` namespaces

Thanks guys!

### CODE REVIEW

File: docs/middleware/index.md [hunk 1]
```diff
@@ -7,16 +7,99 @@ next_link: ./list
...
+conn = Faraday.new do |f|
+  # add by symbol, lookup from Faraday::Request registry
+  f.request :url_encoded
+  # add by symbol, lookup from Faraday::Response registry
+  f.response :follow_redirects
+  # add by symbol, lookup from Faraday::Adapter registry
+  f.adapter :httpclient
+
+  # or
+
```
Comment (olleolleolle) [inline docs/middleware/index.md:74]: Oh, I think th example should be complete alternatives, fully typed out, one after the other. The formatting of th "or..." I would expect to be in regular body text and the code samples with syntax highlighting.

File: docs/middleware/index.md [hunk 2]
```diff
@@ -7,16 +7,99 @@ next_link: ./list
...
+```
+
+### Available Middleware
+
+The [Awesome Faraday](https://github.com/lostisland/awesome-faraday/) project
+has a complete list of useful, well-maintained Faraday middleware. Middleware is
+often provided by external gems, like the
+[faraday-middleware](https://github.com/lostisland/faraday_middleware) gem.
+
+We also have [great documentation](list) for the middleware that ships with
```
Comment (olleolleolle) [inline docs/middleware/index.md:97]: Minor: does this link work? As a casual reader on my phone, I can't recall if we have a "list" link or not.

**Review by olleolleolle — COMMENTED** [review]
Great start of getting the documentation to a place which would have helped "old you" of two weeks ago! Thanks for that.

File: docs/middleware/index.md (hunk 2 above)
Comment (gurgeous) [reply docs/middleware/index.md:98]: Yes, but thank you for asking. I appreciate the attention to detail! I am using `jekyll serve --livereload` to test. It links to the Available Middleware page.

**Review by olleolleolle — APPROVED** [review]
This PR improves the reader's chances of grasping what Connection is in the context of Faraday. Thanks!

**Comment by olleolleolle** [comment]
@gurgeous Thanks! I'll merge this change!

**Pull request merged**